app/controllers/sessions_controller.rb

```diff
@@ -90,8 +90,15 @@ def redirect_to_origin_or_default
   end
 
   def login(options = {})
+    uid = auth_hash.uid
+    if ENV['AUTH_LDAP'] && ENV['USE_LDAP_UID_AS_EXTERNAL_ID']
```

Comment:
I think this might fail if a user allows multiple auth methods (edge-case) and this ... so for example a github login comes in and then ldap blows up ... but I guess nobody uses that, so not really an issue

Reply:
This is a good catch.  I can fix that by checking the auth_hash.provider:
https://github.com/omniauth/omniauth/wiki/Auth-Hash-Schema#schema-10-and-later